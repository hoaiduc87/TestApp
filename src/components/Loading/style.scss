@import 'assets/styles/variables.scss';

.loading {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: $loading-index;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: $white;

  &--transparent {
    background-color: transparent;
  }

  &.hidden {
    visibility: hidden;
    opacity: 0;
  }
}