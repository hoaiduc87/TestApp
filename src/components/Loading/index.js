import React, { memo } from 'react';
import PropTypes from 'prop-types';

import LoadingIcon from 'assets/images/loading.gif';

import './style.scss';

function Loading({
  loading,
  className,
  width,
}) {
  return (
    <div className={`loading ${className || ''} ${!loading ? 'hidden' : ''}`}>
      <img src={LoadingIcon} alt="loading" width={width || '100px'} />
    </div>
  )
}

Loading.propTypes = {
  loading: PropTypes.bool,
  className: PropTypes.string,
  width: PropTypes.any,
};

export default memo(Loading);
