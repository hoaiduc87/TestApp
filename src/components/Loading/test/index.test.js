import React from 'react';
import { render } from '@testing-library/react';

import Loading from '..';

describe('<Loading/>', () => {
  it('Should run and match snapshot', () => {
    const {
      container: { firstChild }
    } = render(<Loading />);
    expect(firstChild).toMatchSnapshot();
  });
})