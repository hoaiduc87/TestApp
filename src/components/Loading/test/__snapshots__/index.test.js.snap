// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> Should run and match snapshot 1`] = `
<div
  class="loading  hidden"
>
  <img
    alt="loading"
    src="loading.gif"
    width="100px"
  />
</div>
`;
